src/agent/Core/ApplicationPool/Group/InternalUtils.cpp

```diff
@@ -325,6 +335,33 @@ Group::callAbortLongRunningConnectionsCallback(const ProcessPtr &process) {
 	}
 }
 
+void
+Group::timeoutRequestsCallback() {
+	boost::shared_ptr<Group> extraReferenceToMe = shared_from_this();
+	if (options.maxRequestQueueTime == 0) {
+		return;
+	}
+	while (!this_thread::interruption_requested()) {
+		syscalls::sleep(options.maxRequestQueueTime);
+		for (deque<GetWaiter>::iterator it = getWaitlist.begin(); it != getWaitlist.end();) {
+			const GetWaiter &waiter = *it;
+			posix_time::time_duration diff = boost::posix_time::microsec_clock::local_time() - waiter.startTime;
+			if (!OXT_LIKELY(!testTimeoutRequestQueue()
+				&& (options.maxRequestQueueTime == 0
+					|| diff.total_seconds() < options.maxRequestQueueTime))) {
+				boost::this_thread::disable_interruption di;
+				boost::this_thread::disable_syscall_interruption dsi;
+				waiter.callback.call(waiter.callback, SessionPtr(), boost::make_shared<RequestQueueTimeoutException>(options.maxRequestQueueSize));
+				it = getWaitlist.erase(it);
+				boost::this_thread::restore_syscall_interruption rsi(dsi);
```

Comment:
This is not necessary. `disable_interruption` and the like are RAII objects. When you leave the scope, the disable flag goes away.